api/pkg/service/user/user.go

```diff
@@ -128,3 +128,56 @@ func createChecksum(token string) string {
 	hash := sha256.Sum256([]byte(token))
 	return hex.EncodeToString(hash[:])
 }
+
+// Refreshes the refresh token of User
+func (s *service) NewRefreshToken(ctx context.Context, p *user.NewRefreshTokenPayload) (*user.NewRefreshTokenResult, error) {
+
+	user, err := s.User(ctx)
```

Comment:
I think from line 135 to 151 we can create a common function and use this from both access token and refresh token since that part of code is duplicated

Reply:
+1 rest looks good